src/utils/storage.ts

```diff
@@ -56,6 +65,59 @@ export class SessionStorage<Params extends unknown[]>
   }
 }
 
+export class LocalForage<Params extends unknown[], Data = string>
+  implements AsyncStorage<Params, Data>
+{
+  constructor(private readonly nameGetter: (...params: Params) => string) {}
+
+  get(...params: Params) {
+    try {
+      if (typeof window === 'undefined') return null
+      return appStorage.getItem<Data>(this.nameGetter(...params))
+    } catch {
+      return null
+    }
+  }
+  set(value: Data, ...params: Params) {
+    try {
+      if (typeof window === 'undefined') return
+      appStorage.setItem(this.nameGetter(...params), value)
+    } catch {}
+  }
+  remove(...params: Params) {
+    try {
+      if (typeof window === 'undefined') return
+      appStorage.removeItem(this.nameGetter(...params))
+    } catch {}
+  }
+}
+
+export class LocalStorageAndForage<Params extends unknown[]>
+  implements Storage<Params>
+{
+  private localStorage: LocalStorage<Params>
+  private localForage: LocalForage<Params>
+
+  constructor(nameGetter: (...params: Params) => string) {
+    this.localStorage = new LocalStorage(nameGetter)
+    this.localForage = new LocalForage(nameGetter)
+  }
+
+  get(...params: Params) {
+    return this.localStorage.get(...params)
+  }
+
+  set(value: string, ...params: Params) {
+    this.localStorage.set(value, ...params)
+    this.localForage.set(value, ...params)
+  }
+
+  remove(...params: Params) {
+    this.localStorage.remove(...params)
+    this.localForage.remove(...params)
```

Comment:
Why you send it to both?

Reply:
Maybe just use forage for all?

Reply:
I use both because forage getItem needs to be awaited
that's why for getting it uses localStorage, but setting uses both, without awaiting the set in forage